node/connection.js

```diff
@@ -213,6 +223,26 @@ TChannelConnection.prototype.onCallResponse = function onCallResponse(res) {
     req.responseEvent.emit(req, res);
 };
 
+TChannelConnection.prototype.onPingRequest = function onPingRequest(req) {
+    var self = this;
+    req.remoteAddr = self.remoteName;
```

Comment:
you shouldn't use an event emitter to mutate the thing being observed, this needs reconsidered.